de.metas.serviceprovider/src/main/java/de/metas/serviceprovider/issue/IssueRepository.java

```diff
@@ -0,0 +1,135 @@
+		record.setAD_User_ID(NumberUtils.asInt(issueEntity.getAssigneeId(), -1));
+		record.setC_Project_ID(NumberUtils.asInt(issueEntity.getProjectId(), -1));
+
+		record.setName(issueEntity.getName());
+		record.setValue(issueEntity.getSearchKey());
+		record.setDescription(issueEntity.getDescription());
+
+		record.setIssueType(issueEntity.getType().getValue());
+		record.setIsEffortIssue(issueEntity.isEffortIssue());
+		record.setProcessed(issueEntity.isProcessed());
+
+		record.setS_Milestone_ID(NumberUtils.asInt(issueEntity.getMilestoneId(), -1));
+		record.setEstimatedEffort(issueEntity.getEstimatedEffort());
+		record.setBudgetedEffort(issueEntity.getBudgetedEffort());
+		record.setEffort_UOM_ID(issueEntity.getEffortUomId().getRepoId());
+
+		record.setExternalId(issueEntity.getExternalIssueId());
+		record.setExternalIssueNo(issueEntity.getExternalIssueNo());
+		record.setIssueURL(issueEntity.getExternalIssueURL());
+
+		InterfaceWrapperHelper.saveRecord(record);
+
+		final IssueId issueId = IssueId.ofRepoId(record.getS_Issue_ID());
+
+		issueEntity.setIssueId(issueId);
+
+		externalIssueDetailsRepository.persistIssueDetails(issueId, issueEntity.getExternalIssueDetails());
+	}
+
+	@NonNull
+	public Optional<IssueEntity> getEntityByExternalId(@NonNull final String externalIssueId)
+	{
+		final I_S_Issue budget_issue = queryBL
+				.createQueryBuilder(I_S_Issue.class)
+				.addOnlyActiveRecordsFilter()
+				.addEqualsFilter(I_S_Issue.COLUMNNAME_ExternalId, externalIssueId)
+				.create()
+				.firstOnly(I_S_Issue.class);
+
+		return budget_issue != null ? Optional.of(buildIssueEntity(budget_issue)) : Optional.empty();
+	}
+
+	@NonNull
+	private IssueEntity buildIssueEntity(final I_S_Issue record)
+	{
+		final Optional<IssueType> issueType = IssueType.getTypeByValue(record.getIssueType());
+
+		if (!issueType.isPresent())
+		{
+			throw new AdempiereException("Unknown IssueType!").appendParametersToMessage()
+					.setParameter("I_S_Issue", record);
+		}
+
+		return IssueEntity.builder()
+				.orgId(OrgId.ofRepoId(record.getAD_Org_ID()))
+				.projectId(ProjectId.ofRepoId(record.getC_Project_ID()))
+				.issueId(IssueId.ofRepoId(record.getS_Issue_ID()))
+				.effortUomId(UomId.ofRepoId(record.getEffort_UOM_ID()))
+				.milestoneId(MilestoneId.ofRepoIdOrNull(record.getS_Milestone_ID()))
+				.assigneeId(UserId.ofRepoIdOrNullISystem(record.getAD_User_ID()))
+				.name(record.getName())
+				.searchKey(record.getValue())
+				.description(record.getDescription())
+				.type(issueType.get())
+				.processed(record.isProcessed())
+				.isEffortIssue(record.isEffortIssue())
+				.estimatedEffort(record.getEstimatedEffort())
+				.budgetedEffort(record.getBudgetedEffort())
+				.externalIssueId(record.getExternalId())
+				.externalIssueNo(record.getExternalIssueNo())
+				.externalIssueURL(record.getIssueURL())
+				.externalIssueDetails(externalIssueDetailsRepository.getByIssueId(IssueId.ofRepoId(record.getS_Issue_ID())))
```

Comment:
@teosarca if you see this, we are aware of the n+1 problem, but chose not to talked it this time. if mass-import (which happens very rarely) becomes too slow, we can still put a staging service in between.
@pvpurcarcosmin please make sure to have an index on the issue-detail's issue-id column